vault/logical_system.go

```diff
@@ -1112,6 +1108,27 @@ func (b *SystemBackend) handleMount(ctx context.Context, req *logical.Request, d
 		}
 	}
 
+	switch version {
+	case "":
+		var err error
+		version, err = selectPluginVersion(ctx, b.System(), logicalType, consts.PluginTypeSecrets)
```

Comment:
Currently no, registering a versioned plugin with the same name overrides the built-in in the same way that registering an unversioned plugin of the same name does. However, I'm wondering whether it might be nice to also allow specifying `-plugin-version=builtin` or `-plugin-version=1.12.0+builtin` to explicitly select the builtin.